build_tools/cmake/iree_check_test.cmake

```diff
@@ -308,7 +308,17 @@ _FILTER_ARCH:_TARGET_CPU_FEATURES. Got: ${_INPUT_TARGET_CPU_FEATURES}")
   # TARGET_CPU_FEATURES_VARIANT is of the form _FILTER_ARCH:_TARGET_CPU_FEATURE.
   list(GET _COMPONENTS 0 _FILTER_ARCH)
   list(GET _COMPONENTS 1 _TARGET_CPU_FEATURES)
-  if(_FILTER_ARCH STREQUAL CMAKE_SYSTEM_PROCESSOR)
+  # Canonicalize the architecture name - some architectures have multiple
```

Comment:
I wonder if we should do some canonicalization at the top level so every place that checks this doesn't have to repeat this logic (that would be a separate PR though).

Reply:
Was thinking the same. Definitely error prone and going to haunt us again. I might send something if no one gets there first :-)